optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,623 @@
+        **kwargs,
+    ):
+        """
+        Load a model and its configuration file from a directory or the HF Hub.
+        Implements: https://github.com/huggingface/huggingface_hub/blob/e67de48368bc1843e40afc1cc9d236402b9609ee/src/huggingface_hub/hub_mixin.py#L73
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+            use_auth_token (`str` or `bool`):
+                The token to use as HTTP bearer authorization for remote files. Needed to load models from a private
+                repository.
+            revision (`str`):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id.
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (`bool`, *optional*, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            encoder_file_name(`str`, *optional*):
+                The encoder model file name overwriting the default file name, allowing to save the encoder model with
+                a different name.
+            decoder_file_name(`str`, *optional*):
+                The decoder model file name overwriting the default file name, allowing to save the decoder model with
+                a different name.
+            decoder_with_past_file_name(`str`, *optional*):
+                The decoder with past key values model file name overwriting the default file name, allowing to save
+                the decoder model with a different name.
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization.
+        """
+        config_dict = kwargs.pop("config", {})
+        config = PretrainedConfig.from_dict(config_dict)
+        encoder_file_name = encoder_file_name or ONNX_ENCODER_NAME
+        decoder_file_name = decoder_file_name or ONNX_DECODER_NAME
+        decoder_with_past_file_name = decoder_with_past_file_name or ONNX_DECODER_WITH_PAST_NAME
+
+        # Load model from a local directory
+        if os.path.isdir(model_id):
+            model = cls.load_model(
+                encoder_path=os.path.join(model_id, encoder_file_name),
+                decoder_path=os.path.join(model_id, decoder_file_name),
+                decoder_with_past_path=os.path.join(model_id, decoder_with_past_file_name),
+                config=config,
+            )
+            kwargs["model_save_dir"] = Path(model_id)
+            kwargs["encoder"] = encoder_file_name
+            kwargs["decoder"] = decoder_file_name
+            kwargs["decoder_with_past"] = decoder_with_past_file_name
```

Comment:
To match what we have for encoder/decoder only models
```suggestion
            kwargs["last_encoder_name"] = encoder_file_name
            kwargs["last_decoder_name"] = decoder_file_name
            kwargs["last_decoder_with_past_name"] = decoder_with_past_file_name
```